### PR TITLE
DHIS2-2683 Remove callbacks from the `onEnter` handlers

### PR DESCRIPTION
**Pull request by varl**
Due to how HashHistory is implemented when it changes URL hashes
in v3 of react-router multiple events fires when we use
`HashHistory.push(url)`. This in turn causes the `onEnter` handler
in react-router to run twice.

The XHR request itself is cached so no extra data is transmitted from
the server but another consequence is that the components render twice
which has performance implications.

For more information see the [JIRA
issue](https://jira.dhis2.org/browse/DHIS2-2683).

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -138,56 +141,48 @@ function createLoaderForSchema(schema, actionCreatorForLoadingObject, resetActiv
...
+        // changed by adding an organisation unit which would need to be
+        // reflected in the organisation unit tree
         initState({ params });
-        return callback();
     }
 
     initState({ params });
     return listActions.loadList(params.modelType)
         .take(1)
         .subscribe(
```
Comment (varl) [inline src/router.js]: I'm pretty new to RxJS so I'm not sure if it makes sense to subscribe here anymore to capture the error messages, but I added a `noop` function to allow our error handling to function as it did before.

Comment (Birkbjo) [reply src/router.js]: I believe it does, as the loadList, and similar functions are still async (but handled by observables) and can throw an error, so this makes sense to me!

File: src/router.js [hunk 2]
```diff
@@ -17,6 +17,8 @@ import { loadProgramIndicator } from './EditModel/program-indicator/actions';
 
 import onDemand from './on-demand';
 
+function noop () {}
```
Comment (Birkbjo) [inline src/router.js]: Could probably get this from lodash?

Comment (varl) [reply src/router.js]: Aight, fixed in 5ecac0d.

**Pull request merged**